test/java/seedu/addressbook/commands/SortCommandTest.java

```diff
@@ -0,0 +1,92 @@
+package seedu.addressbook.commands;
+
+import org.junit.Before;
+import org.junit.Test;
+import seedu.addressbook.data.AddressBook;
+import seedu.addressbook.data.person.*;
+import seedu.addressbook.util.TestUtil;
+
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+public class SortCommandTest {
+
+    private AddressBook emptyAddressBook;
+    private AddressBook addressBook;
+    private AddressBook alreadySortedAddressBook;
+
+    private List<ReadOnlyPerson> emptyDisplayList;
+    private List<ReadOnlyPerson> sortedListWithEveryone;
+
+    @Before
+    public void setUp() throws Exception {
+        Person johnDoe = new Person(new Name("John Doe"), new Phone("61234567", false),
+                new Email("john@doe.com", false), new Address("395C Ben Road", false), Collections.emptySet());
+        Person janeDoe = new Person(new Name("Jane Doe"), new Phone("91234567", false),
+                new Email("jane@doe.com", false), new Address("33G Ohm Road", false), Collections.emptySet());
+        Person samDoe = new Person(new Name("Sam Doe"), new Phone("63345566", false),
+                new Email("sam@doe.com", false), new Address("55G Abc Road", false), Collections.emptySet());
+        Person davidGrant = new Person(new Name("David Grant"), new Phone("61121122", false),
+                new Email("david@grant.com", false), new Address("44H Define Road", false),
+                Collections.emptySet());
+
+        emptyAddressBook = TestUtil.createAddressBook();
+        addressBook = TestUtil.createAddressBook(johnDoe, janeDoe, davidGrant, samDoe);
+        alreadySortedAddressBook = TestUtil.createAddressBook(davidGrant, janeDoe, johnDoe, samDoe);
+
+        emptyDisplayList = TestUtil.createList();
+
+        sortedListWithEveryone = TestUtil.createList(davidGrant, janeDoe, johnDoe, samDoe);
+    }
+
+    @Test
+    public void execute_sort() {
+        assertSortSuccessful(emptyAddressBook, emptyDisplayList);
+        assertSortSuccessful(addressBook, sortedListWithEveryone);
+        assertSortSuccessful(alreadySortedAddressBook, sortedListWithEveryone);
+    }
+
+    /**
+     * Creates a new sort command.
+     */
+    private SortCommand createSortCommand(AddressBook addressBook,
+                                          List<ReadOnlyPerson> displayList) {
+
+        SortCommand command = new SortCommand();
+        command.setData(addressBook, displayList);
+
+        return command;
+    }
+
+    /**
+     * Executes the command, and checks that the execution was what we had expected.
+     */
+    private void assertCommandBehaviour(SortCommand sortCommand, String expectedMessage,
```

Comment:
This test is general enough to be applied to any command. Maybe consider modularising it in the future.

Reply:
Good idea. I think we should do this in another PR. What do you think?

Reply:
you can have the setup method in testUtil so that it can be used for other tests